Integrations/integration-McAfeeNSM.yml

```diff
@@ -0,0 +1,679 @@
+        var res = request("GET", cmd_url, headers, '');
+
+        // store current credentials
+        session_credentials = Base64.encode(res.session + ':' + res.userId);
+
+        return res;
+    }
+
+    function get_attacks(attack_id) {
+        var cmd_url, res;
+        if (attack_id) {
+            if (attack_id.match('^0x[0-9A-Fa-f]{8}$') === null) {
+                return 'Error! Attack ID must be formated as 32-bit hexadecimal number. for example: 0x1234BEEF';
+            }
+
+            cmd_url = 'attack/{0}'.format(attack_id);
+        } else {
+            cmd_url = 'attacks';
+        }
+
+        res = request('GET', cmd_url, {}, '');
+        if (attack_id) {
+            return attacks_to_entry('Attack {0}'.format(attack_id), [res.AttackDescriptor]);
+        } else {
+            return attacks_to_entry('{0} Attacks'.format(res.AttackDescriptorDetailsList.length), res.AttackDescriptorDetailsList);
+        }
+    }
+
+    function get_IPS_policies(domain_id) {
+        var cmd_url = 'domain/{0}/ipspolicies'.format(domain_id);
+        var res = request('GET', cmd_url, {}, '');
+
+        return IPS_policies_to_entry('{0} IPS Policies:'.format(res.PolicyDescriptorDetailsList.length), res.PolicyDescriptorDetailsList);
+    }
+
+    function get_IPS_policy_details(policy_id) {
+        var cmd_url = 'ipspolicy/{0}'.format(policy_id);
+        var res = request('GET', cmd_url, {}, '');
+
+        return IPS_policy_to_entry('IPS Policy {0}'.format(policy_id), res.PolicyDescriptor, policy_id);
+    }
+
+    function get_sensors(domain_id) {
+        var cmd_url = 'sensors';
+        if (domain_id !== undefined) {
+            cmd_url += '?domain={0}'.format(domain_id);
+        }
+        var res = request('GET', cmd_url, {}, '');
+
+        return sensors_to_entry('Sensors', res.SensorDescriptor);
+    }
+
+    function get_domains(domain_id) {
+        var cmd_url = 'domain';
+        if (domain_id) {
+            cmd_url += '/{0}'.format(domain_id);
+        }
+        // var res = request('GET', cmd_url, {}, JSON.stringify({'domain_id' : domain_id}));
+        //log(JSON.stringify(res));
+        var res = request('GET', cmd_url, {}, '');
+
+        if (domain_id) {
+            return domains_to_entry('Domain {0}:'.format(domain_id), [res]);
+        } else {
+            return domains_to_entry('{0} Domains:'.format(res.length || ''), [res]);
+        }
+    }
+
+    function all_alerts_url(state, time_period, start_time, end_time, search, filter) {
+        var cmd_url = 'alerts';
+        var cmd_args = {};
+
+        if (state) {
+            cmd_args.alertstate = state;
+        }
+
+        if (time_period) {
+            cmd_args.timeperiod = time_period;
+            if (time_period === 'CUSTOM') {
+                if ((start_time === undefined) || (end_time === undefined)) {
+                    return 'Error! In CUSTOM mode, You must specify both start time and end time';
```

Comment:
you don't do anything this error.
recommended options:
1. throw error
2. return empty string (`''`) and validate in caller 
3. don't validate this here, just write something like `function validateTimeArgs(time_period, start_time, end_time) {..}` and call it in the relevant places in the `switch` in the bottom of the script.